Userland/Libraries/LibC/unistd.cpp

```diff
@@ -975,8 +975,41 @@ int unveil(char const* path, char const* permissions)
 // https://pubs.opengroup.org/onlinepubs/9699919799/functions/getpass.html
 char* getpass(char const* prompt)
 {
-    dbgln("FIXME: getpass('{}')", prompt);
-    TODO();
+    int tty = open("/dev/tty", O_RDWR | O_NOCTTY | O_CLOEXEC);
+
+    if (tty < 0)
+        return nullptr;
+
+    struct termios backup, noecho;
+    tcgetattr(tty, &backup);
+
+    noecho = backup;
+    noecho.c_lflag &= ~(ECHO|ISIG|INLCR|IGNCR);
+    noecho.c_lflag |= ICANON | ICRNL;
+
+    tcsetattr(tty, TCSAFLUSH, &noecho);
+    tcdrain(tty);
+
+    write(tty, prompt, strlen(prompt));
+
+    static char password[PASS_MAX];
+    ssize_t chars_read = read(tty, password, sizeof(password));
+
+    tcsetattr(tty, TCSAFLUSH, &backup);
+
+    write(tty, "\n", sizeof("\n"));
```

Comment:
`sizeof("\n")` == 2, was it intended to write 2 bytes with a null terminator or just the new line?